floris/tools/wind_data.py

```diff
@@ -0,0 +1,529 @@
+
+        # Price table
+        if self.price_table is not None:
+            self.price_table_flat = self.price_table.flatten()
+        else:
+            self.price_table_flat = None
+
+        # Set mask to non-zero frequency cases depending on compute_zero_freq_occurence
+        if self.compute_zero_freq_occurence:
+            # If computing zero freq occurences, then this is all True
+            self.non_zero_freq_mask = [True for i in range(len(self.freq_table_flat))]
+        else:
+            self.non_zero_freq_mask = self.freq_table_flat > 0.0
+
+        # N_findex should only be the calculated cases
+        self.n_findex = np.sum(self.non_zero_freq_mask)
+
+    def unpack(self):
+        """
+        Unpack the flattened versions of the matrices and return the values
+        accounting for the non_zero_freq_mask
+        """
+
+        # The unpacked versions start as the flat version of each
+        wind_directions_unpack = self.wd_flat.copy()
+        wind_speeds_unpack = self.ws_flat.copy()
+        freq_table_unpack = self.freq_table_flat.copy()
+
+        # Now mask thes values according to self.non_zero_freq_mask
+        wind_directions_unpack = wind_directions_unpack[self.non_zero_freq_mask]
+        wind_speeds_unpack = wind_speeds_unpack[self.non_zero_freq_mask]
+        freq_table_unpack = freq_table_unpack[self.non_zero_freq_mask]
+
+        # Repeat for turbulence intensity if not none
+        if self.ti_table_flat is not None:
+            ti_table_unpack = self.ti_table_flat[self.non_zero_freq_mask].copy()
+        else:
+            ti_table_unpack = None
+
+        # Now get unpacked price table
+        if self.price_table_flat is not None:
+            price_table_unpack = self.price_table_flat[self.non_zero_freq_mask].copy()
+        else:
+            price_table_unpack = None
+
+        return (
+            wind_directions_unpack,
+            wind_speeds_unpack,
+            freq_table_unpack,
+            ti_table_unpack,
+            price_table_unpack,
+        )
+
+    def resample_wind_rose(self, wd_step=None, ws_step=None):
+        """
+        Resamples the wind rose by by wd_step and/or ws_step
+
+        Args:
+            wd_step: Step size for wind direction resampling (float, optional).
+            ws_step: Step size for wind speed resampling (float, optional).
+
+        Returns:
+            WindRose: Resampled wind rose based on the provided or default step sizes.
+
+        Notes:
+            - Returns a resampled version of the wind rose using new `ws_step` and `wd_step`.
+            - Uses the bin weights feature in TimeSeries to resample the wind rose.
+            - If `ws_step` or `wd_step` is not specified, it uses the current values.
+        """
+        if ws_step is None:
+            if len(self.wind_speeds) >= 2:
+                ws_step = self.wind_speeds[1] - self.wind_speeds[0]
+            else:
+                # It doesn't matter, just set to 1
```

Comment:
Some more detail here would improve the readability and help in understanding the logic. What doesn't matter and why doesn't it matter?

Reply:
Yes that's better, thanks